### DISCUSSION TITLE
UR5 manipulator with drl_grasping 2.0.0

### DISCUSSION THREAD

**Discussion by lucarei** [AndrejOrsula/drl_grasping#107] [General]
Hi Andrej. 

I am working on a pick and place application for pick up trash and clean beaches. I have a similar setup to yours: a manipulator (one of the family of Universal Robots) on top of a mobile base (Robotnik).

Then, I should use UR manipulators in this project but i red this on 2.0.0 changelog:
> REMOVED: ur5_rg2 and kinova_j2s7s300 from supported robot models due to shift to ros2_control.

Is it possible to use again Universal Robot's arms or is forbidden for this version of drl_grasping? If it is possible, could you share with me some tips/hints to do it?

**Comment by AndrejOrsula** [comment]
Hello,

In `2.0.0`, this repository supports robots that implement [`gz_ros2_control` (renamed from `ign_ros2_control`)](https://github.com/ros-controls/gz_ros2_control). Therefore, it is not "forbidden" to use the current version of `drl_grasping` with any robot - it is just not implemented.

In versions `1.x.y`, [JointTrajectoryController plugin of Ignition](https://github.com/AndrejOrsula/ign_moveit2_examples/tree/legacy_jtc_ign_plugin) was used - but that was before `gz_ros2_control` and stable API of `ros2_control`. The repository for UR5 robot with RG2 used in `1.x.y` is still available under https://github.com/AndrejOrsula/ur5_rg2_ign, but I archived it as I do not have access to this robot anymore. 

If you want to start from the `ur5_rg2_ign` repository, you are free to fork it and update it to use `ign_ros2_control` such that it matches https://github.com/AndrejOrsula/panda_ign_moveit2 or https://github.com/snt-spacer/lunalab_summit_xl_gen. More specifically, this PR (https://github.com/AndrejOrsula/panda_ign_moveit2/pull/7) contains all the changes that would need to be made (but adjusted for the specific robot). It is a lot of changes, but you could use it as a template if you don't find any other workaround.

After that, the UR robot would just need to be added to the [models](https://github.com/AndrejOrsula/drl_grasping/tree/master/drl_grasping/envs/models/robots). The model for this robot is already in `1.x.y` as `ur5_rg2.py` ([here](https://github.com/AndrejOrsula/drl_grasping/blob/1.1.0/drl_grasping/envs/models/robots/ur5_rg2.py)), so you could just copy it and make minor modifications to match the other robots.

**Comment by lucarei** [comment]
Thank you!
